### ISSUE TITLE
Syntax constructor Inspired on Windows Network Policy Server

### ISSUE THREAD

**Issue by fischerdouglas** [tacacsgui/tacacsgui#85]
(Not a Bug, a Wish.)

TL;DR,
My wish/suggestion is a tool that would be inspired in that modeling to create an open-source tool to replace NPS.

I believe that you know NPS - Windows Network Policy Server.

I worked with several tools for AAA, Cisco, Huawei...
But none of then were so intuitive as Windows NPS to construct the syntax of:
 - The match-conditions
 - Defining the constraints
 - Defining additional settings.

I believe that this occurs for three reasons:
a) The concepts of the objects are very well defined, a goodly presented operator.
b) The dictionaries on many-many information, Vendors AV-Pairs, etc... Is very well populated and maintained.
c) The Split between Connection Policies and Network Request Policies usually is misunderstood, but very-very much powerful when you have many different databases, and when you need to do Radius-Proxy to external Radius.

NPS covers only Radius protocol.
But I want to believe that that same modeling is pretty much applicable to TACACS also.

I would like to see something that could replace NPS, including TACACS(with integrated Polices between Tacacas and Radius) on a tool that could be used as GUI, and also as an API-controlled tool.

If the community and developers get interested in that idea, we could discuss more profoundly the goals, and also some crowdfunding to opensource that.
